spec/acceptance/qpid_tool_spec.rb

```diff
@@ -15,11 +15,28 @@
     end
   end
 
+  context 'removing qpid-tools' do
+    let(:pp) do
+      <<-PUPPET
+      class { 'qpid::tools':
+        ensure => 'absent',
+      }
+      PUPPET
+    end
+
+    it_behaves_like 'a idempotent resource'
+
+    describe package('qpid-tools') do
+      it { is_expected.not_to be_installed }
+    end
+  end
+
   context 'installing qpid' do
     let(:pp) do
       <<-PUPPET
-      class { 'qpid::tools': } ->
-      class { 'qpid': }
+      class { 'qpid':
+        require => Class['qpid::tools'],
```

Comment:
Why is this require needed? Looking at it, it feels like `qpid::config_cmd` should have this relationship. You can also use `contain qpid::tools` in `qpid::install` to ensure the relationship. Both is also an option.

Reply:
Note that this is in the acceptance tests. The idea with the test is to ensure that #46 is no longer a problem by first installing the qpid-tools package and then installing the qpid packages.

Reply:
But what are you testing? That if you manually provide a workaround that it doesn't show up? If you provide the correct chaining (the current patch does **not** do that), the catalog should fail to compile if you first try to install tools before the client, right?

Reply:
I am missing what is being seen as the manual work around.

The issue posed in #46 was: if you install qpid-tools before installing qpid-cpp-server, the qpid-tools package will install a 'compat' package that then blocks the installation of qpid-cpp-server. This is no longer true. You can currently install them in whatever order you want awithout one breaking the other.

The idea behind the test was to ensure that qpid-tools can be installed first, and that qpid-cpp-server can be installed afterward without running into not being able to install qpid-cpp-server due to the previous, no longer existent, compat package. To prove that we no longer needed this relationship:

https://github.com/theforeman/puppet-qpid/commit/734b44b8dd2061743f0b12695482ac691682deb5#diff-29fdc8c170b6278c50ae7a6359737e386dc4769ce94b9a6e505f5bcd3e4890deL9

Reply:
I pushed a 2nd commit which adds a comment above the test case, explaining the reason for the `require => Class['qpid::tools'],` in the test case and linking to https://github.com/theforeman/puppet-qpid/issues/46

Reply:
> The issue posed in #46 was: if you install qpid-tools before installing qpid-cpp-server, the qpid-tools package will install a 'compat' package that then blocks the installation of qpid-cpp-server. This is no longer true. You can currently install them in whatever order you want awithout one breaking the other.

Can you tell me how you know it's fixed? Is there no chance of an older repository still having the broken versions? I'm mostly asking because it is in EPEL7 and we have less control over that.

That's also something I really dislike about combining things in a commit. I assumed it was a regression in 734b44b8dd2061743f0b12695482ac691682deb5 because there was nothing mentioned about it at all. However, it is a big change that deserved a PR on its own. That would have informed me that it was intentional and not a side effect. My instinct now was to write patch to revert part of it.

Reply:
> Can you tell me how you know it's fixed? Is there no chance of an older repository still having the broken versions? I'm mostly asking because it is in EPEL7 and we have less control over that.

Sure an old repository could have some old version of this package and break. But that shouldn't be a supported case. We test and expect the current versions in EPEL.

In my original change, my intent was to allow the ensuring of qpid the server. As #46 is no longer a problem, and I did not want to combine ensuring qpid-tools and ensuring qpid-server it made sense to me to break the dependency (again assuming #46 was no longer an issue which it wasnt). Could I have tried to do this as separate changes? Yes. Did it occur to us at the time? No. The question now is how do we move forward with this change. Are there changes requested here?

Reply:
Then I would suggest to be explicit in the test:
```puppet
include qpid
# Old versions of qpid-tools installed compat-qpid-cpp-client causing a conflict
# https://github.com/theforeman/puppet-qpid/issues/46
# This verifies it does work now
Package['qpid-tools'] -> Package['qpid-cpp-server', 'qpid-cpp-client']
```

However, to really ensure it's a clean system then you also ensure the server packages are not present. Otherwise you're not really reproducing what was a problem.

You could also argue that its more important to test that the default invocation without any additional code is more important. We can't reproduce the problematic situation anyway (since EPEL no longer ships broken packages) so what the current code is really trying to achieve is some regression testing.

I think I'd drop this test altogether since I don't think it's really testing anything anymore.

Reply:
I've opened https://github.com/theforeman/puppet-qpid/pull/160 to drop the test. If that is satisfactory then I'll rebase this one.